### DISCUSSION TITLE
Python library wanted: vporter (akin nimporter) - directly import V code from Python

### DISCUSSION THREAD

**Discussion by dumblob** [vlang/v#17110] [Ideas]
Anyone works on a [nimporter](https://github.com/Pebaz/nimporter )-like project for V?

I think that would be really awesome as many Pythoners are lazy :wink: and probably tend to prefer the C syntax (and compilation speed!) over the comprehensive Nim syntax (with lower compilation speed).

**Comment by MatejMagat305** [comment]
I think that is only about porting Python.h to v and call similar
https://stackoverflow.com/questions/42393369/c-importing-python-module
And I do not know whether somebody work on this

**Comment by Wajinn** [comment]
I think the situation with Nim is a bit different.  V is a C-like language, where Nim is a Python-like language and attempts to attract that audience.  If there are any V users/developers so very deep into Python, then maybe they should create a module that does light transpiling (Python rules to V rules) or implements brace elision (where use of "{...}" becomes optional/unneeded).

**Reply by dumblob** [reply]
That is an interesting observation. I absolutely did not have in mind anything like this :wink:.

My use case is simple - one will need to write Python (for whatever reason) or more frequently already has some python code base. One wants to either slowly transition to *V* (i.e. chunk-wise rewrite the whole code base over a few years) or wants to implement certain routines in a highly performant way.

In these cases one has to learn the Python C API which is a huge annoyance (it is also easy to shoot yourself in the foot - most such modules I have seen in companies **deliberately** ignore garbage collection, multithreading, multiprocessing, etc. - in other words it is a pure horror) and the whole integration (compiling the module manually, waiting minutes for all the compilation & tests to run, distributing it, maintaining it with Python C API changes etc.) is another huge annoyance.

Therefore Nim guys want to write the modules in Nim and use them "transparently" without any of the hassles I outlined.

And I would like to do the same with *V* :wink:.
